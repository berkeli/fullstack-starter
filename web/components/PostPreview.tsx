import { Box, Heading, Text } from "@chakra-ui/react";
import React from "react";

type Props = {
  post: any;
};

const PostPreview = ({ post }: Props) => {
  return (
    <Box p="4">
      <Heading as="h3">{post.title}</Heading>
      <Text>{post.content}</Text>
    </Box>
  );
};

export default PostPreview;
