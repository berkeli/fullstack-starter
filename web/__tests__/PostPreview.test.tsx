import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import PostPreview from "../components/PostPreview";

describe("PostPreview", () => {
  it("should render correctly", () => {
    const { container } = render(<PostPreview post={{ title: "test", content: "test" }} />);
    expect(container).toMatchSnapshot();
  });
});
