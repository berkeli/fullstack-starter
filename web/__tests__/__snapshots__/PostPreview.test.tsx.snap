// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostPreview should render correctly 1`] = `
<div>
  <div
    class="css-2rlxtj"
  >
    <h3
      class="chakra-heading css-0"
    >
      test
    </h3>
    <p
      class="chakra-text css-0"
    >
      test
    </p>
  </div>
</div>
`;
